partcad/src/partcad/part_factory_enrich.py

```
@@ -89,13 +101,43 @@ def __init__(self, ctx, source_project, target_project, config):
-            if "with" in config:
-                for param in config["with"]:
-                    augmented_config["parameters"][param]["default"] = config["with"][param]
-            orig_source_project.init_part_by_config(
-                augmented_config,
-                source_project,
-            )
+            if "with" in part.config:
+                if "parameters" not in augmented_config:
+                    raise Exception(
+                        "Attempting to parametrize a part that has no parameters: %s" % str(augmented_config)
+                    )
```

Ruff (0.8.2)

Create your own exception

(TRY002)